File: LeetCode/704_BinarySearch/binarySearch.ts
function search(nums: number[], target: number): number {
    if (nums.length === 0 || target < nums[0] || target > nums[nums.length - 1]) {
        return -1;
    }

    let left = 0;
    let right = nums.length - 1;
  
    while (left <= right) {
        let mid = left + Math.floor((right - left) / 2);
        if (nums[mid] === target) return mid;
        else if (nums[mid] > target) right = mid - 1;
        else left = mid + 1;
    }
    return -1;
}

File: LeetCode/704_BinarySearch/binarySearch.py


class Solution:
    def search(self, nums, target):
        left,right = 0, len(nums)-1
        while left <= right:
            mid = left + (right-left)//2
            if target == nums[mid]: return mid
            elif target < nums[mid]: right = mid - 1
            else: left = mid + 1
        return -1
    

sol = Solution()

print(sol.search([-1,0,3,5,9,12], 9))